packages/components/src/components/Table/Table.stories.js

```diff
@@ -0,0 +1,234 @@
+      <Table
+        showTitle={boolean('showTitle', false)}
+        resource={text('resource', 'Resource Name')}
+        rows={rows}
+        headers={[
+          { key: 'name', header: 'Name' },
+          { key: 'namespace', header: 'Namespace' },
+          { key: 'date', header: 'Date Modified' }
+        ]}
+        selectedNamespace="*"
+        loading={boolean('loading', false)}
+        toolbarButtons={[
+          { handler: action('handleNew'), text: 'Add', icon: Add }
+        ]}
+      />
+    );
+  })
+  .add('table with one row, 1 batchAction button and checkboxes', () => {
+    const rows = [
+      {
+        id: 'namespace1:resource-one',
+        name: text('Name', 'resource-one'),
+        namespace: text('Namespace', 'namespace1'),
+        date: text('Date', '100 years ago')
+      }
+    ];
+
+    return (
+      <Table
+        showTitle={boolean('showTitle', false)}
+        resource={text('resource', 'Resource Name')}
+        rows={rows}
+        headers={[
+          { key: 'name', header: 'Name' },
+          { key: 'namespace', header: 'Namespace' },
+          { key: 'date', header: 'Date Modified' }
+        ]}
+        selectedNamespace="*"
+        loading={boolean('loading', false)}
+        batchActionButtons={[
+          { handler: action('handleDelete'), text: 'Delete', icon: Delete }
+        ]}
+      />
+    );
+  })
+  .add(
+    'table with one sortable row, 2 batchAction and 2 toolbar buttons checkboxes',
+    () => {
+      const rows = [
+        {
+          id: 'namespace1:resource-one',
+          name: 'resource-one',
+          namespace: 'namespace1',
+          date: '100 years ago'
+        },
+        {
+          id: 'default:resource-two',
+          name: 'resource-two',
+          namespace: 'default',
+          date: '2 weeks ago'
+        },
+        {
+          id: 'tekton:resource-three',
+          name: 'resource-three',
+          namespace: 'tekton',
+          date: '2 minutes ago'
+        }
+      ];
+
+      return (
+        <Table
+          showTitle={boolean('showTitle', false)}
+          resource={text('resource', 'Resource Name')}
+          rows={rows}
+          headers={[
+            { key: 'name', header: 'Name' },
+            { key: 'namespace', header: 'Namespace' },
+            { key: 'date', header: 'Date Modified' }
+          ]}
+          selectedNamespace="*"
+          sortable={boolean('sortable', true)}
+          batchActionButtons={[
+            { handler: action('handleDelete'), text: 'Delete', icon: Delete },
+            { handler: action('handleRebuild'), text: 'Rebuild', icon: Rebuild }
+          ]}
+          toolbarButtons={[
+            { handler: action('handleNew'), text: 'Add', icon: Add },
+            {
+              handler: action('handleRebuildAll'),
+              text: 'RebuildAll',
+              icon: RebuildAll
+            }
+          ]}
+        />
+      );
+    }
+  );
+
+/*
+  HOW TO USE:
+  - Required Props:
```

Comment:
these should be documented on the component itself using `Table.propTypes` https://reactjs.org/docs/typechecking-with-proptypes.html

can specify required props using `isRequired`, default is optional

we can also provide default values for the optional props if this is useful using `Table.defaultProps`